src/cache-utils.ts

```diff
@@ -95,3 +96,32 @@ export const getCacheDirectoryPath = async (
 
   return stdOut;
 };
+
+function logWarning(message: string): void {
+  const warningPrefix = '[warning]';
+  core.info(`${warningPrefix}${message}`);
+}
+
+export function isGhes(): boolean {
+  const ghUrl = new URL(
+    process.env['GITHUB_SERVER_URL'] || 'https://github.com'
+  );
+  return ghUrl.hostname.toUpperCase() !== 'GITHUB.COM';
+}
+
+export function isCacheFeatureAvailable(): boolean {
+  if (!cache.isFeatureAvailable()) {
+    if (isGhes()) {
+      logWarning(
```

Comment:
I would prefer here to throw an error instead of a warning. Because this keeps it consistent with other setup-* actions https://github.com/actions/setup-java/pull/308/files#diff-3294a832ea2276e554177e0b3007cc2d401c082912c7fbde49fa09141bf1aed1R87-R103

Reply:
Done. Thank you.